### PR TITLE
Use GPU with snapml

### PR DESCRIPTION
**Pull request by tanglef**
Same as logreg l2

| rcv1 | news20 |
| ----- | ----------- |
| Capture d’écran de 2022-05-09 20-54-34 |Capture d’écran de 2022-05-09 20-54-28 |

### CODE REVIEW

File: solvers/snapml.py [hunk 1]
```diff
@@ -13,13 +16,15 @@ class Solver(BaseSolver):
 
     def set_objective(self, X, y, lmbd):
         self.X, self.y, self.lmbd = X, y, lmbd
+        self.use_gpu = cuda_version is not None
```
Comment (agramfort) [inline solvers/snapml.py]: can you make GPU a parameter of the Solver so we can run both in one benchmark? you should just skip the parameter gpu=True when cuda is not available. Please also do this on the Lasso benchmark. 🙏

**Comment by mathurinm** [comment]
Works fine without gpu on my machine: 
image

I don't have the same results though, sklearn is faster...

**Review by mathurinm — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @tanglef !

**Comment by tanglef** [comment]
@mathurinm 
Because the main point of this solver is to use the GPU (when it makes sense to do so)
That's when the `GPU` as solver parameter is interesting (thanks for that recommendation @agramfort, btw)
See how with/without gpu on `rcv1` the results change:

Capture d’écran de 2022-05-10 09-04-30

**Pull request merged**